services/_shared/personas/code_reviewer/complexity.py

```
@@ -0,0 +1,184 @@
+_DEFAULT_COGNITIVE_CAP = int(os.getenv("GRUG_COMPLEXITY_COGNITIVE_CAP", "25"))
+
+_RULE = "high-complexity"
+
+
+def _changed_lines(hunk: DiffHunk) -> set[int]:
+    """New-side line numbers ADDED in this hunk (unified-diff walk, mirrors
+    sast._added_lines but returns only the numbers)."""
+    out: set[int] = set()
+    lineno = hunk.new_start
+    for raw in hunk.body.splitlines():
+        if raw.startswith("@@") or raw.startswith("+++") or raw.startswith("---"):
```

Ruff (0.15.20)

[warning] Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)